### PR TITLE
Implemented Reply for a infallible result

### PR DESCRIPTION
**Pull request by comath**
`http::Error` implements `From<Infallible>`, this just shortcuts that so we can have an easy time with infallible results.

Another option is to implement the following:
```
impl<T,E> Reply for Result<T, E>
where
    T: Reply + Send,
    E: Reply + Send,
{ ... }
```

### CODE REVIEW

File: src/reply.rs [hunk 1]
```diff
@@ -427,6 +427,23 @@ where
...
+{
+    #[inline]
+    fn into_response(self) -> Response {
+        match self {
+            Ok(t) => t.into_response(),
+            Err(e) => {
+                // Maybe panic?
+                tracing::error!("Infallible Error: {:?}", e);
+                StatusCode::INTERNAL_SERVER_ERROR.into_response()
+            }
```
Comment (MikailBag) [inline src/reply.rs:442]: If I understand correctly, this should work better
```suggestion
            Err(e) => match e {}
```

Comment (comath) [reply src/reply.rs]: I did not know that about `Infallible`, I changed it to what you suggested.

**Comment by kaj** [comment]
For reference: The "other option", `impl Reply for Result<T, E> where T: Reply, E: Reply`, is suggested in #458 .

**Comment by comath** [comment]
> For reference: The "other option", `impl Reply for Result<T, E> where T: Reply, E: Reply`, is suggested in #458 .

I should have looked around some more, sorry :). I thought it'd be much much more complicated than this simple addition.

**Review by jxs — APPROVED** [review]

**Pull request closed without merging**